scrapy/utils/response.py

```diff
@@ -60,7 +67,7 @@ def response_httprepr(response):
     return b"".join(values)
 
 
-def open_in_browser(response, _openfunc=webbrowser.open):
+def open_in_browser(response: Response, _openfunc: Callable[[str], Any] = webbrowser.open) -> Any:
```

Comment:
Nitpick: return type is not Any, but whatever _openfunc returns. TypeVar would be more precise here.

Reply:
Makes sense, but I think I need a bit of assistance. If I do:
```python
OpenInBrowserTypeVar = TypeVar("OpenInBrowserTypeVar")

def open_in_browser(
    response: Union[HtmlResponse, TextResponse],
    _openfunc: Callable[[str], OpenInBrowserTypeVar] = webbrowser.open,
) -> OpenInBrowserTypeVar:
```
I get
```
error: Incompatible default for argument "_openfunc" (default has type "Callable[[str, int, bool], bool]", argument has type "Callable[[str], OpenInBrowserTypeVar]")
```
because [`webbrowser.open`](https://docs.python.org/3/library/webbrowser.html) takes `url, new=0, autoraise=True`. However, the function is called with `_openfunc(f"file://{fname}")`, passing only a `str`. Of course I could do
```
_openfunc: Callable[[str], OpenInBrowserTypeVar] = webbrowser.open,  # type: ignore
```
but I'd like to avoid that.

I haven't found a solution so far, my next step would be to ask in SO, but maybe you know a way to solve it.